lib/rspec/matchers/built_in/include.rb

```diff
@@ -13,34 +13,50 @@ def initialize(*expected)
         # @return [Boolean]
         def matches?(actual)
           @actual = actual
-          perform_match(:all?, :all?)
+          if actual.respond_to?(:include?)
+            @divergent_items = excluded_from_actual
+            @divergent_items.empty?
+          else
+            false
+          end
         end
 
         # @api private
         # @return [Boolean]
         def does_not_match?(actual)
           @actual = actual
-          perform_match(:none?, :any?)
+          if actual.respond_to?(:include?)
+            @divergent_items = included_from_actual
+            @divergent_items.empty?
+          else
+            false
+          end
         end
 
         # @api private
         # @return [String]
         def description
-          described_items = surface_descriptions_in(expected)
-          item_list = EnglishPhrasing.list(described_items)
-          improve_hash_formatting "include#{item_list}"
+          improve_hash_formatting("include#{readable_list_of(expected)}")
         end
 
         # @api private
         # @return [String]
         def failure_message
-          improve_hash_formatting(super) + invalid_object_message
+          if actual.respond_to?(:include?)
+            improve_hash_formatting("expected #{description_of @actual} to include#{readable_list_of @divergent_items}")
+          else
+            improve_hash_formatting(super) + invalid_object_message
+          end
```

Comment:
TBH I am not a fan of this (as well as it's duplication with the `when_negated` version). Any thoughts on how to clean/combine this up?